### PR TITLE
Tpetra: Fix #5457; fix #5458

### LINKED ISSUE #5457: Tpetra: Finite-element assembly example broke deprecated OFF build
## Bug Report
@trilinos/tpetra 

### Description

```
In file included from .../packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_main.cpp:61:0: 
.../packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_TotalElementLoop_SP.hpp: In function ‘int TpetraExamples::executeTotalElementLoopSPKokkos_(const Teuchos::RCP<const Teuchos::Comm<int> >&, const TpetraExamples::CmdLineOpts&)’: 
.../packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_TotalElementLoop_SP.hpp:568:28: error: cannot call member function ‘void Kokkos::Serial::fence() const’ without object    
execution_space_t::fence();
                             ^ 
.../packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_TotalElementLoop_SP.hpp:601:28: error: cannot call member function ‘void Kokkos::Serial::fence() const’ without object
    execution_space_t::fence();
                             ^

```

Fix is to call `fence` as an instance method, not a class (static) method.

### LINKED ISSUE #5458: Tpetra: Tpetra_FECrsGraph_decl.hpp does not include forward decl header
## Bug Report
@trilinos/tpetra @csiefer2 

### Description

`Tpetra_FECrsGraph_decl.hpp` does not include the forward declaration header file `Tpetra_FECrsGraph_fwd.hpp`.  This means that `Tpetra::FECrsGraph` does not realize that it has default template parameters.

### PR DESCRIPTION
**Pull request by mhoemmen**
@trilinos/tpetra 

## Description

Fix #5457 and #5458.  These relate to Tpetra's finite-element assembly examples.

## Related Issues

* Closes #5457, #5458

### CODE REVIEW

File: packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_InsertGlobalIndices_FE_SP.hpp [hunk 1]
```diff
@@ -233,52 +236,58 @@ int executeInsertGlobalIndicesFESP_(const Teuchos::RCP<const Teuchos::Comm<int>
...
+      ReferenceQuad4(element_matrix);
+      ReferenceQuad4RHS(element_rhs);
 
-      for(size_t col_idx=0; col_idx<4; col_idx++)
-      {
-        column_scalar_values[col_idx] = element_matrix(element_node_idx, col_idx);
+      // Fill the global column ids array for this element
+      for (size_t element_node_idx=0;
+           element_node_idx < owned_element_to_node_ids.extent(1);
+           ++element_node_idx) {
```
Comment (csiefer2) [inline packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_InsertGlobalIndices_FE_SP.hpp]: You know, modern monitors support more than 80 columns... :tongue:

Comment (mhoemmen) [reply packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_InsertGlobalIndices_FE_SP.hpp]: @csiefer2 Seriously though, if you're visually impaired and have to blow up the font, long lines are super annoying

Comment (mhoemmen) [reply packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_InsertGlobalIndices_FE_SP.hpp]: but thanks for the review :-D

Comment (csiefer2) [reply packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_InsertGlobalIndices_FE_SP.hpp]: Point taken.  Just had to share the love, though :stuck_out_tongue:

**Comment by csiefer2** [comment]
WIP'ing to wait for: https://github.com/mhoemmen/Trilinos/pull/1

File: packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_InsertGlobalIndices_FE_SP.hpp (hunk 1 above)
Comment (kddevin) [reply packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_InsertGlobalIndices_FE_SP.hpp]: Clearly @mhoemmen  has seen the font on my laptop (probably from all the way across the big conference room).  Thanks!

**Comment by mhoemmen** [comment]
Hold on, I got a test failure.  Let me try again.

**Comment by mhoemmen** [comment]
Never mind; false alarm.  I'll push my changes and take off the WIP label.

**Comment by mhoemmen** [comment]
@csiefer2 ready now :-D

**Comment by csiefer2** [comment]
> @csiefer2 ready now :-D

By your command :robot:

**Comment by csiefer2** [comment]
Re-WIPing due to https://github.com/mhoemmen/Trilinos/pull/2

**Comment by mhoemmen** [comment]
Thanks @csiefer2 !  I just merged your changes into my fork's branch; ready for rereview :-)
